js/main.js

```diff
@@ -1,22 +1,109 @@
+
+const descriptionRoom = 'Двухместный комфортабельный номер, с кондиционером, душем и парковкой';
+
+const ROOM_PHOTOS = [
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/duonguyen-8LrGtIxxa4w.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/brandon-hoogenboom-SNxQGWxZQi0.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/claire-rendall-b6kAwr1i0Iw.jpg'
+];
 
-// https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Math/random ссылка на источник!
+const locationLat = {1: 35.65000, 2: 35.70000, 3: 5};
 
-function getRandomFloating (min, max, maxDigits = 0) {
-  if (min > max || min < 0 || max <= 0) {
-    return ('Неверный диапазон значений');
+const locationLng = {1: 139.70000, 2: 139.80000, 3: 5};
+
+const COUNT_ITEMS_OF_OFFER = 10;
+
+/* Старые функции убрал, взял новые у кекса, хотя не уверен в правильности, те немного поправил после твоих заметок
+и они как то локаничней смотрелись!*/
+
+const getRandomPositiveInteger = (a, b) => {
+  const lower = Math.ceil(Math.min(Math.abs(a), Math.abs(b)));
+  const upper = Math.floor(Math.max(Math.abs(a), Math.abs(b)));
+  const result = Math.random() * (upper - lower + 1) + lower;
+  return Math.floor(result);
+};
+getRandomPositiveInteger(1, 8);
+
+const getRandomPositiveFloat = (a, b, digits = 1) => {
+  const lower = Math.min(Math.abs(a), Math.abs(b));
+  const upper = Math.max(Math.abs(a), Math.abs(b));
+  const result = Math.random() * (upper - lower) + lower;
+  return +result.toFixed(digits);
+};
+getRandomPositiveFloat(1.3, 4.9, 3);
+
+const getRandomArrayElement = (elements) => elements[getRandomPositiveInteger(0, elements.length - 1)];
+
+const makeRandomSelection = (count) => {
+  const newFeatures = [];
+  if (count >= FEATURES_ITEMS.length) {
+    return FEATURES_ITEMS;
   }
-  const degree = 10 ** maxDigits;
-  return ~~ ((Math.random() *(max - min) + min) * degree) / degree;
-}
-getRandomFloating(1, 4, 3);
+  for (let i = 0; i < count; i++) {
+    let newElement = FEATURES_ITEMS[Math.floor(Math.random() * FEATURES_ITEMS.length)];
+    while (newFeatures.includes(newElement)) {
+      newElement = FEATURES_ITEMS[Math.floor(Math.random() * FEATURES_ITEMS.length)];
+    }
+    newFeatures.push(newElement);
+  }
+  return newFeatures;
+};
+
+
+const makeTitles = () => ({
+  author: {
+    avatar: '',//Тут не понимаю как можно сделать
+  },
+  offer: {
+    title: offerTitle,
+    adress: ' ',// здесь тоже провал
+    price: getRandomArrayElement(PRICE_ROOM),
+    type: getRandomArrayElement(TYPE_OF_ROOMS),
+    rooms: getRandomArrayElement(ROOMS_COUNT),
+    guests: getRandomArrayElement(GUESTS_COUNT),
+    checkin: getRandomArrayElement(CHEK_IN_COUNTER),
+    checkout: getRandomArrayElement(CHECK_OUT_COUNTER),
+    features: makeRandomSelection(3), // здесь я тоже не уверен, нашел функцию в интернете!
```

Comment:
В целом функция работает правильно, но если это конкретно функция для особенностей (features) то лучше назвать её getRandomFeatures(3) а ещё лучше унифицировать эту функцию, добавив второй парамтер, например
```suggestion
    features: makeRandomSelection(3, FEATURES_ITEMS),
```
И заменить в теле функции все FEATURE_ITEMS на второй аргумент